src/Effect/Exec.hs

```diff
@@ -145,14 +146,13 @@ instance (Algebra sig m, MonadIO m) => Algebra (Exec :+: sig) (ExecIOC m) where
       let mangleResult :: (ExitCode, Stdout, Stderr) -> Either CmdFailure Stdout
           mangleResult (exitcode, stdout, stderr) =
             case (exitcode, cmdAllowErr cmd) of
-                  (ExitSuccess, _) -> Right stdout
-                  (_, Never) -> Left $ mkFailure exitcode stdout stderr
-                  (_, NonEmptyStdout) ->
-                    if BL.null stdout
-                      then Left $ mkFailure exitcode stdout stderr
-                      else Right stdout
-                  (_, Always) -> Right stdout
-
+              (ExitSuccess, _) -> Right stdout
```

Comment:
a lot of unrelated formatting changes in this commit; continuing to do formatting with `ormolu` piecemeal as I edit files